sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c

```diff
@@ -866,37 +866,36 @@ static int sof_card_dai_links_create(struct device *dev,
 			      dmic_component, 1, dmic_init, NULL);
 	}
 
-#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_COMMON_HDMI_CODEC)
+#if IS_ENABLED(CONFIG_SND_HDA_CODEC_HDMI)
 	/* HDMI */
-	if (hdmi_num > 0) {
-		idisp_components = devm_kcalloc(dev, hdmi_num,
-						sizeof(*idisp_components),
+	for (i = 0; i < hdmi_num; i++) {
+		struct snd_soc_dai_link_component *idisp_components;
+
+		idisp_components = devm_kzalloc(dev, sizeof(*idisp_components),
```

Comment:
Why do you want to allocate them one by one? What was wrong with allocating the whole array at once?

Reply:
@lyakh  First this array is defined out of #ifdef  HDMI_CODE  but used here. So if HDMI is not defined, I got a build warning message. I don't want to add anther #ifdef HDMI for definition of this array, so I moved the definition into this loop.

Reply:
@RanderWang so you prefer to run 4 kmalloc calls with error handling in the likely case of HDMI being used (in about 100% of the cases) to having one more #ifdef? That doesn't seem like a good choice to me. #ifdefs are frowned upon in the Linux kernel in .c files, that's true, but we have a ton of them around... And you can certainly find many ways to avoid that additional #ifdef. In fact you can avoid them completely here. You can use the `if (hdmi_num > 0)` clause above, but that would add indentation. Or you can make a new function. Either way choosing code aesthetics and sacrificing run-time doesn't seem a good solution to me in this case.